main.nf

```diff
@@ -1072,49 +1073,107 @@ pairBam = bamNormal.cross(bamTumor).map {
-        set file(genomeFile), file(genomeIndex), file(genomeDict), file(dbsnp), file(dbsnpIndex) from Channel.value([
+        set idPatient, idSampleNormal, file(bamNormal), file(baiNormal), idSampleTumor, file(bamTumor), file(baiTumor), file(intervalBed) from pairBamMutect2
+        set file(genomeFile), file(genomeIndex), file(genomeDict), file(intervals), file(germlineResource), file(germlineResourceIndex) from Channel.value([
             referenceMap.genomeFile,
             referenceMap.genomeIndex,
             referenceMap.genomeDict,
-            referenceMap.dbsnp,
-            referenceMap.dbsnpIndex
+            referenceMap.intervals,
+            referenceMap.germlineResource,
+            referenceMap.germlineResourceIndex
         ])
 
     output:
-        set val("MuTect2"), idPatient, val("${idSampleTumor}_vs_${idSampleNormal}"), file("${intervalBed.baseName}_${idSampleTumor}_vs_${idSampleNormal}.vcf") into vcfMuTect2
+        set val("Mutect2"), 
+            idPatient,
+            val("${idSampleTumor}_vs_${idSampleNormal}"),
+            file("${intervalBed.baseName}_${idSampleTumor}_vs_${idSampleNormal}.vcf") into mutect2Output
+        set idPatient,
+            idSampleTumor,
+            idSampleNormal,
+            file("${intervalBed.baseName}_${idSampleTumor}_vs_${idSampleNormal}.vcf.stats") optional true into mutect2Stats
 
     when: 'mutect2' in tools
 
     script:
+    // please make a panel-of-normals, using at least 40 samples
+    // https://gatkforums.broadinstitute.org/gatk/discussion/11136/how-to-call-somatic-mutations-using-gatk4-mutect2
+    PON = params.pon ? "--panel-of-normals $params.pon" : ""
+
     """
+    # Get raw calls
+    # this case we are getting raw calls only for the intervals, we also have to concatenate them
     gatk --java-options "-Xmx${task.memory.toGiga()}g" \
-        Mutect2 \
-        -R ${genomeFile}\
-        -I ${bamTumor}  -tumor ${idSampleTumor} \
-        -I ${bamNormal} -normal ${idSampleNormal} \
-        -L ${intervalBed} \
-        -O ${intervalBed.baseName}_${idSampleTumor}_vs_${idSampleNormal}.vcf
+      Mutect2 \
+      -R ${genomeFile}\
+      -I ${bamTumor}  -tumor ${idSampleTumor} \
+      -I ${bamNormal} -normal ${idSampleNormal} \
+      -L ${intervalBed} \
+      --germline-resource ${germlineResource} \
+      ${PON} \
+      -O ${intervalBed.baseName}_${idSampleTumor}_vs_${idSampleNormal}.vcf
     """
 }
 
-vcfMuTect2 = vcfMuTect2.groupTuple(by:[0,1,2])
+mutect2Output = mutect2Output.groupTuple(by:[0,1,2,3])
+(mutect2Output, mutect2OutForStats) = mutect2Output.into(2)
 
-// STEP FREEBAYES
+(mutect2Stats,intervalStatsFiles) = mutect2Stats.into(2)
+mutect2Stats = mutect2Stats.groupTuple(by:[0,1,2])
+
+process MergeMutect2Stats {
+    tag {idSampleTumor + "_vs_" + idSampleNormal}
+
+    publishDir "${params.outdir}/VariantCalling/${idSampleTumor}_vs_${idSampleNormal}/Mutect2", mode: params.publishDirMode
+
+    input:
+        set caller,
```

Comment:
I don't think you have `caller` here in this channel.

Reply:
You could also group the two sets into just one with another `groupTuple`

Reply:
Now there is the missing caller - I would prefer to leave it it as it is as there is a fork later and it is already pretty cluttered.